### DISCUSSION TITLE
Overlayfs support added to Lxroot

### DISCUSSION THREAD

**Discussion by parke** [parke/lxroot#24] [Announcements]
In June, I happened to read the following on the [`user_namespaces(7)`](https://www.man7.org/linux/man-pages/man7/user_namespaces.7.html) man page.

```
   Effect of capabilities within a user namespace

       [snip...]

       Holding CAP_SYS_ADMIN  within  the  user  namespace
       that  owns  a process's mount namespace allows that
       process to create bind mounts and mount the follow‐
       ing types of filesystems:

           •  [snip...]
           •  overlayfs (since Linux 5.11)
```

This was an exciting discovery!  My anticipated use case is:

1.  Mounting (as root) a read-only squashfs filesystem, for use as a `newroot`
2.  Overlaying (via Lxroot as non-root) a writable filesystem on top of that read-only squashfs `newroot`.
3.  And then being able to **efficiently make** arbitrary changes to the overlaid `newroot`.
4.  And also being able to **efficiently undo** all those changes by deleting the `upperdir` and `workdir`.

Pursuant to the above goal, today I added initial overlayfs support to Lxroot.  If anyone is interested in using Lxroot's new overlayfs support, please contact me and I will start working towards publishing a new release of Lxroot.

For more information about overlayfs, please see the [`mount(8)`](https://man7.org/linux/man-pages/man8/mount.8.html) man page.

**Comment by fyrak1s** [comment]
That sounds thrilling...! I am currently clearing my backlog in order to get to this..
Will try to give it a go on GoboLinux :) but no haste, it will take me probably 1-2 months before I am gonna get to this.

Edit: Maybe this would also be useful to create an encrypted directory?
